actor/builtin/miner/miner.go

```diff
@@ -905,9 +902,22 @@ func (ma *Actor) SubmitPoSt(ctx exec.VMContext, poStProof types.PoStProof, fault
 		//
 		// This switching will be removed when issue #2270 is completed.
 		if !ma.Bootstrap {
-			seed, err := getPoStChallengeSeed(ctx, state)
-			if err != nil {
-				return nil, errors.RevertErrorWrap(err, "failed to sample chain for challenge seed")
+			// calculate challenge seed now. If PoSt is too early, this will fail.
+			poStChallengeTime := types.NewBlockHeight(PoStChallengeTimeBlocks)
+
+			var seed types.PoStChallengeSeed
+			if chainHeight.LessThan(state.ProvingPeriodEnd) {
+				// proof is in time, sample at start of post challenge time
+				seed, err = getPoStChallengeSeed(ctx, state, state.ProvingPeriodEnd.Sub(poStChallengeTime))
+				if err != nil {
+					return nil, errors.RevertErrorWrap(err, "failed to sample chain for challenge seed, PoSt is probably early")
```

Comment:
Include the window open height, window end, and current chain height in the message.